dask_expr/_expr.py

```diff
@@ -2248,9 +2248,11 @@ def _fusion_pass(expr):
 
                 group.append(next)
                 for dep in dependencies[next]:
-                    if (dep.npartitions == root.npartitions) and not (
-                        dependents[dep] - set(stack) - set(group)
-                    ):
+                    if (
+                        dep.npartitions == root.npartitions
+                        or dep.npartitions == 1
+                        or root.npartitions == 1
+                    ) and not (dependents[dep] - set(stack) - set(group)):
```

Comment:
```suggestion
                    if (
                        dep.npartitions == root.npartitions
                        or next._broadcast_dep(dep)
                    ) and not (dependents[dep] - set(stack) - set(group)):
```
Your logic makes sense, but I think we can simply check if `dep` is a "broadcastable" dependency of `next`.